### PR TITLE
added Q.POP, Q.FND and Q.REM

### PR DESCRIPTION
**Pull request by attejensen**
#### What does this PR do?
Adds three new OPs
Q.POP: pop first element or element at index from list (stack)
Q.FND: return index of first element with value x, -1 if not found
Q.REM: remove first element with value x, return index of removed element, -1 of none removed

#### How should this be manually tested?
Q.GRW 1 // enable grow
Q.CLR 2 // put 2 on stack
Q 4         // put 4 on stack
Q 6         // put 6 on stack
Q 8         // put 8 on stack, now stack is | 8 6 4 2 |
Q.POP   // pop first element, will return 8, stack is now | 6 4 2 |
Q.POP 1 // pop element at index 1, will return 4, stack is now | 6 2 |
Q.FND 1  // find element with value 1, none is found returns -1
Q.FND 2  / find element with value 2, found at index 1, return 1, stack is unchanged
Q.REM 2 // remove first element with value 2, return index of removed element, will return 1, stack is now | 6 |
Q.REM 1 // remove first element with value 1, not found, will return -1, stack is unchanged

#### Any background context you want to provide?
This makes it possible to use Q as stack. Searching and removing elements by value are handy tools.

#### I have,
* [x] updated `CHANGELOG.md` and `whats_new.md`
* [x] updated the documentation
* [x] updated `help_mode.c` (if applicable)
* [x] run `make format` on each commit
* [x] run tests

### CODE REVIEW

File: module/help_mode.c [hunk 1]
```diff
@@ -107,6 +107,9 @@ const char* help2[HELP2_LENGTH] = { "2/16 VARIABLES",
                                     "Q|SHIFT REGISTER",
                                     "Q.N|SET Q LENGTH",
                                     "Q.AVG|AVERAGE OF ALL Q",
+				    "Q.POP|POP FIRST ELEMENT",
```
Comment (scanner-darkly) [inline module/help_mode.c:110]: indentation is off

**Comment by scanner-darkly** [comment]
the tests are failing on `Q.POP`

**Comment by tehn** [comment]
wondering if it's worth the effort to implement the tests within github actions

**Comment by attejensen** [comment]
Will rewrite ASAP.

**Pull request closed without merging**